### PR TITLE
Approve/Reject for pending versions only

### PR DESCRIPTION
**Pull request by wagnerand**
Make sure only pending versions can be approved or rejected.

### CODE REVIEW

**Comment by eviljeff** [comment]
break all the tests! 👎

**Comment by eviljeff** [comment]
If fixing the existing tests doesn't specifically test for this functionality you should add a new one.

**Comment by wagnerand** [comment]
Yea, I know. On it...

**Comment by wagnerand** [comment]
Also, I blame the tests for not setting statuses correctly and therefore testing invalid scenarios!

File: src/olympia/constants/base.py [hunk 1]
```diff
@@ -415,12 +415,10 @@
 
 # Types of Canned Responses for reviewer tools.
 CANNED_RESPONSE_ADDON = 1
-CANNED_RESPONSE_APP = 2
 CANNED_RESPONSE_PERSONA = 3
 
 CANNED_RESPONSE_CHOICES = {
     CANNED_RESPONSE_ADDON: _('Add-on'),
-    CANNED_RESPONSE_APP: _('App'),
```
Comment (eviljeff) [inline src/olympia/constants/base.py]: Are these required changes?  They look unconnected.  (If they are connected please add a migration to clear out the app canned responses)

File: src/olympia/editors/tests/test_forms.py [hunk 2]
```diff
@@ -45,47 +47,71 @@ def test_other_statuses(self):
             if status in statuses:
                 return
             else:
-                label = self.set_status(status)['prelim']['label']
+                status = self.set_statuses(addon_status=status,
```
Comment (eviljeff) [inline src/olympia/editors/tests/test_forms.py]: lets not redeclare `status` to be something completely different.

File: src/olympia/editors/tests/test_forms.py [hunk 3]
```diff
@@ -45,47 +47,71 @@ def test_other_statuses(self):
...
-        reject = self.set_status(amo.STATUS_LITE)['reject']['details']
+        status = self.set_statuses(addon_status=amo.STATUS_LITE,
+                                   file_status=amo.STATUS_UNREVIEWED)
+        reject = status['reject']['details']
         assert force_unicode(reject).startswith('This will reject the files')
 
     def test_not_public(self):
-        # If the file is unreviewed then there is no option to reject,
-        # so the length of the actions is one shorter
-        assert len(self.set_status(amo.STATUS_UNREVIEWED)) == 5
```
Comment (eviljeff) [inline src/olympia/editors/tests/test_forms.py]: Was the old comment completely wrong or has this test changed what its doing?

File: src/olympia/constants/base.py (hunk 1 above)
Comment (wagnerand) [inline src/olympia/constants/base.py]: This is just cleanup. `CANNED_RESPONSE_APP` was only used in that one test and not in actual code.

File: src/olympia/editors/tests/test_forms.py (hunk 3 above)
Comment (wagnerand) [inline src/olympia/editors/tests/test_forms.py]: The old comment did not make any sense at all. I changed it to what the test is actually testing.

File: src/olympia/editors/tests/test_forms.py (hunk 2 above)
Comment (wagnerand) [inline src/olympia/editors/tests/test_forms.py]: Ugh...

**Comment by wagnerand** [comment]
Updated the patch.

File: src/olympia/constants/base.py (hunk 1 above)
Comment (eviljeff) [inline src/olympia/constants/base.py]: `CANNED_RESPONSE_APP` isn't used directly, but `CANNED_RESPONSE_CHOICES` is used in the `CannedResponse` model.  Anyway, there are canned responses with type=3 still around (dev has them, for example) so if you're removing the last reference to what that id means then you should remove them with a migration.

**Comment by eviljeff** [comment]
r+wc

File: src/olympia/constants/base.py (hunk 1 above)
Comment (eviljeff) [inline src/olympia/constants/base.py]: If possible I'd move this to a separate commit.

Comment (wagnerand) [inline src/olympia/constants/base.py]: Do you mean type=2 ?

Comment (eviljeff) [inline src/olympia/constants/base.py]: I do.

**Comment by wagnerand** [comment]
Thank you!
